python/lsst/summit/utils/peekExposure.py

```diff
@@ -0,0 +1,1026 @@
+
+        mode, binSize, binnedSourceCat = self.runPeek(
+            exposure, mode, donutDiameter
+        )
+
+        table = self.transformTable(binSize, binnedSourceCat)
+
+        wgood = self.getGoodSources(mode, binnedSourceCat)
+
+        # prepare output variables
+        maxFluxIdx, b_centroid, b_shape = self.getBrightest(
+            binnedSourceCat, binSize, wgood
+        )
+        psf_shape = self.getPsfShape(binnedSourceCat, binSize, wgood)
+
+        shapes_NW, shapes_AA = self.transformShapes(
+            [b_shape, psf_shape],
+            exposure,
+            binSize,
+        )
+
+        if doDisplay:
+            self.updateDisplay(
+                exposure, binSize, binnedSourceCat, maxFluxIdx, doDisplayIndices
+            )
+
+        return pipeBase.Struct(
+            mode=mode,
+            binSize=binSize,
+            binnedSourceCat=binnedSourceCat,
+            table=table,
+            brightestIdx=maxFluxIdx,
+            brightestCentroid=b_centroid,
+            brightestShape=b_shape,
+            brightestShapeNW=shapes_NW[0],
+            brightestShapeAA=shapes_AA[0],
+            psfShape=psf_shape,
+            psfShapeNW=shapes_NW[1],
+            psfShapeAA=shapes_AA[1],
+        )
+
+    def runPeek(self, exposure, mode, donutDiameter):
+        """Classify exposure and run appropriate PeekTask wrapper.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure to peek.
+        mode : {'auto', 'donut', 'spec', 'photo'}
+            Mode to run in.
+        donutDiameter : float
+            Donut diameter in pixels.
+
+        Returns
+        -------
+        result : `pipeBase.Struct`
+            Result of the peek.
+            Struct containing:
+                - mode : str
+                    Peek mode that was run.
+                - binSize : int
+                    Binning factor used.
+                - binnedSourceCat : `lsst.afw.table.SourceCatalog`
+                    Source catalog from the binned exposure.
+        """
+        if mode == "auto":
+            # Note, no attempt to handle dispersed donuts.  They'll default to
+            # donut mode.
+            if donutDiameter > self.config.donutThreshold:
+                mode = "donut"
+            elif isDispersedExp(exposure):
+                mode = "spec"
+            else:
+                mode = "photo"
+
+        match mode:
+            case "donut":
+                result = self.donut.run(exposure, donutDiameter)
+            case "spec":
+                result = self.spec.run(exposure)
+                if len(result.binnedSourceCat) == 0:
+                    self.log.warn("No sources found in spec mode.")
+                    if self.config.doPhotoFallback:
+                        self.log.warn("Falling back to photo mode.")
+                        result = self.photo.run(exposure, nobin=True)
```

Comment:
Oh, is this the only reason `nobin` exists? If so then I think it's fine to leave it as-is, and just have the binning controlled by the binning factor then (as people can just use 1 for no binning). Out of interest though, why is this necessarily not binned? Because in this case we're known to be struggling, and therefore our best chance of winning is to go totally unbinned?

Reply:
Oh, I think I remember now.  It's because having passed through self.spec.run, the exposure is already binned, so this is to prevent binning it a second time...

Reply:
Although, the clone() was supposed to prevent that?  I'll have to trace through to be certain...